syntax = "proto3";

package cc;

option go_package = "github.com/RoseWrightdev/Video-Conferencing/backend/go/gen/cc/proto;cc";

service CaptioningService {
  // Bidirectional streaming: SFU sends audio chunks, Service sends back captions.
  rpc StreamAudio(stream AudioChunk) returns (stream CaptionEvent);
}

message AudioChunk {
  string session_id = 1;      // Unique ID for the audio stream/user
  bytes audio_data = 2;       // Raw PCM audio
  string target_language = 3; // Optional: ISO code (e.g., "en", "es"). If empty, defaults to transcription.
}

message CaptionEvent {
  string session_id = 1;
  string text = 2;
  bool is_final = 3;
  double confidence = 4;
}
